### PR TITLE
Final PR

### PR DESCRIPTION
**Pull request by sciabarracom**
This PR includes
- refactored proxy
- source support
- main support
- actionloop image (base image for swift and golang)
- actionloop-golang image (golang runtime)
- images with compilation ability
It supersedes the other images and provides an all-inclusive handover

### CODE REVIEW

**Comment by mrutkows** [comment]
@sciabarracom In addition, you need an intiial LICENSE(.txt) file and NOTICE(.txt) file please.

File: openwhisk/_test/build.sh [hunk 1]
```diff
@@ -1,5 +1,39 @@
 #!/bin/bash
 cd "$(dirname $0)"
+<<<<<<< HEAD
```
[the remark this answers is not in the document]
Comment (sciabarracom) [reply openwhisk/_test/build.sh]: ops.... how it happened? the test were passing... will fix asap

**Comment by sciabarracom** [comment]
yup of course, I thought they were already in place...

**Comment by mrutkows** [comment]
@sciabarracom did you need help adding scancode to Travis?

**Comment by sciabarracom** [comment]
I think I figured out how to do. I have some half done uncommitted work.
I will ask on slack if I have problems thx.
--
  Michele Sciabarra, CEO
  Sciabarra.com ltd
  sciabarracom@sciabarra.com



On Tue, Apr 24, 2018, at 11:46 PM, Matt Rutkowski wrote:
> @sciabarracom[1] did you need help adding scancode to Travis?


> — You are receiving this because you were mentioned. Reply to this
> email directly, view it on GitHub[2], or mute the thread[3].> 


Links:

  1. https://github.com/sciabarracom
  2. https://github.com/apache/incubator-openwhisk-runtime-go/pull/8#issuecomment-384090579

**Comment by dubee** [comment]
@sciabarracom, might be easier to review these changes if they were broken out into multiple PRs.

**Comment by sciabarracom** [comment]
I already did a subset of the PR, opening a branch and no one touched it for weeks, and it became quickly obsolete and it was a nightmare to keep it updated while I was waiting for review.. So I closed it and reopened this one.  Furthermore,  changes to the core are really just a few. The patch looks big because I rewrote all the tests in Go.  Everything with _test in the name are tests. I think it will save a lot of time to everyone if we review this big one once and for all so further changes will be smaller. It is a handover of a runtime, unfortunately. Also the only thing I can really do here to make smaller is to remove the tests (and add them again later). I do not think it is really useful to do that.

**Comment by csantanapr** [comment]
I created an Infra ticket  to enable issues and Travis for the repo
https://issues.apache.org/jira/browse/INFRA-16436

**Comment by sciabarracom** [comment]
To help review I will submit another PR with all the tests removed in order to focus on the important parts, but I will keep this PR.

**Comment by csantanapr** [comment]
@sciabarracom let me know when your done with commits to this branch/PR
So I can review and merge

**Comment by sciabarracom** [comment]
I fixed all the code issues and the scancode is passing. Also the build scripts locally are working the only issue is the travis build is failing (I think because I need to setup the environment better). You can go ahead merging and reviewing.  I am going only to change the travis build scripts to fix the travis  build.

**Comment by csantanapr** [comment]
I can take a look on Travis and help with that

**Comment by sciabarra** [comment]
Ok, FINALLY the travis build is fixed, and the scancode pass. It is all in order now. Please proceed...

**Comment by csantanapr** [comment]
Thanks

**Pull request merged**